### PR TITLE
Add ES5 containers to docker-compose configuration

### PR DESCRIPTION
**Pull request by joxl**
Adds ES5 docker configuration for running tests

Jira: [SAAS-12533](https://dimagi-dev.atlassian.net/browse/SAAS-12533)

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Tests added where ES code change was required.

### QA Plan

No QA.

### Safety story

Majority of this change set is testing-only (docker for travis, etc) code.  The small "HQ" code that required changing now has tests.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/util/es/elasticsearch.py [hunk 1]
```diff
@@ -24,8 +24,27 @@
...
+        NotFoundError,
+        SerializationError,
+        TransportError,
+        RequestError,
+    )
+    from elasticsearch5.client import (
+        IndicesClient,
+        SnapshotClient,
+    )
+    from elasticsearch5.helpers import bulk, scan
```
[the remark this answers is not in the document]
Comment (joxl) [reply corehq/util/es/elasticsearch.py:45]: ignore

**Comment by dannyroberts** [comment]
Does this actually change travis to use the new elasticsearch5 docker setup somehow, or is it just giving us the ability to run it locally if we want to?

**Comment by joxl** [comment]
@dannyroberts, It only adds the ability for ES5, which is only used locally at the moment.  It also supplies the ES5 boilerplate, making it trivial to configure ES5 travis jobs if that's something you want to do.

File: requirements/base-requirements.in [hunk 2]
```diff
@@ -44,8 +44,8 @@ django~=2.2.21
 djangorestframework~=3.12.2
 dnspython==1.15.0
 dropbox==9.3.0
-elasticsearch==1.9.0
```
Comment (dannyroberts) [inline requirements/base-requirements.in:47]: nice

File: corehq/ex-submodules/pillowtop/processors/elastic.py [hunk 3]
```diff
@@ -189,12 +189,16 @@ def send_to_elasticsearch(index_info, doc_type, doc_id, es_getter, name, data=No
             if delete:
                 es_interface.delete_doc(alias, doc_type, doc_id)
             else:
-                params = {'retry_on_conflict': 2}
                 if es_merge_update:
-                    es_interface.update_doc_fields(alias, doc_type, doc_id, fields=data, params=params)
+                    # The `retry_on_conflict` param is only valid on `update`
+                    # requests. ES <5.x was lenient of its presence on `index`
+                    # requests, ES >=5.x is not.
```
Comment (dannyroberts) [inline corehq/ex-submodules/pillowtop/processors/elastic.py:195]: but just to triple check, in 2.x it's presence or absence has no effect on anything other than an update request?

**Review by dannyroberts — APPROVED** [review]

File: corehq/ex-submodules/pillowtop/processors/elastic.py (hunk 3 above)
Comment (joxl) [reply corehq/ex-submodules/pillowtop/processors/elastic.py:195]: Correct. In fact I went deep into the git history to figure out why we're even adding it on `index` requests and discovered that the current state is a regression bug. In previous implementations we were only adding it on `update` requests, but that detail got lost in a refactor (possibly during the ES7 changes).

Comment (dannyroberts) [reply corehq/ex-submodules/pillowtop/processors/elastic.py:195]: Awesome, thanks for digging!

**Review by millerdev — APPROVED** [review]

**Pull request merged**